e2etest/declarativeHelpers.go

```diff
@@ -174,26 +176,27 @@ var eOperation = Operation(0)
 type Operation uint8
 
 func (Operation) Copy() Operation        { return Operation(1) }
-func (Operation) Sync() Operation        { return Operation(2) }
-func (Operation) CopyAndSync() Operation { return eOperation.Copy() & eOperation.Sync() }
-func (Operation) Remove() Operation      { return Operation(3) }
+func (Operation) Sync() Operation        { return Operation(1<<1) }
+func (Operation) CopyAndSync() Operation { return eOperation.Copy() | eOperation.Sync() }
+func (Operation) Remove() Operation      { return Operation(1<<2) }
+func (Operation) Resume() Operation 	 { return Operation(1<<3) } // Resume should only ever be combined with Copy or Sync, and is a mid-job cancel/resume.
 
 func (o Operation) String() string {
 	return enum.StringInt(o, reflect.TypeOf(o))
 }
 
 // getValues chops up composite values into their parts
 func (o Operation) getValues() []Operation {
-	switch o {
-	case eOperation.Copy(),
-		eOperation.Sync(),
-		eOperation.Remove():
-		return []Operation{o}
-	case eOperation.CopyAndSync():
-		return []Operation{eOperation.Copy(), eOperation.Sync()}
-	default:
-		panic("unexpected operation type")
+	out := make([]Operation, 0)
+	// separate out the bitflags
+	for idx := 0; idx < 8; idx++ {
+		opMatching := Operation(1<<idx)
+		if opMatching & o != 0 {
+			out = append(out, opMatching)
+		}
 	}
+
+	return out
```

Comment:
Maybe we should still panic if the operation type is incorrect?

Reply:
_Maybe_. The reason I'm hesitant is because of magic numbers.